commit b35497f8f53c559b0ff788f0b7ecafaa7969113e
Author: Tobias Junghans <tobydox@veyon.io>
Date:   Mon Jan 10 22:40:39 2022 +0000

    SafeSocketNotifier: fix socket FD type and remove signal argument
    
    Qt uses qintptr for the socket file descriptor.
    
    The argument is not used by any user of the SafeSocketNotifier class so we can eliminate it to fix a compile error when building for the x86_64-w64-mingw32 architecture caused by the new QSocketNotifier::activated() overload.

diff --git a/src/qca_safeobj.cpp b/src/qca_safeobj.cpp
index 7133990f..b2ad3888 100644
--- a/src/qca_safeobj.cpp
+++ b/src/qca_safeobj.cpp
@@ -34,7 +34,7 @@ inline void releaseAndDeleteLater(QObject *owner, QObject *obj)
     obj->deleteLater();
 }
 
-SafeSocketNotifier::SafeSocketNotifier(int socket, QSocketNotifier::Type type, QObject *parent)
+SafeSocketNotifier::SafeSocketNotifier(qintptr socket, QSocketNotifier::Type type, QObject *parent)
     : QObject(parent)
 {
     sn = new QSocketNotifier(socket, type, this);
diff --git a/src/qca_safeobj.h b/src/qca_safeobj.h
index ad819958..5c47adad 100644
--- a/src/qca_safeobj.h
+++ b/src/qca_safeobj.h
@@ -33,7 +33,7 @@ class SafeSocketNotifier : public QObject
 {
     Q_OBJECT
 public:
-    SafeSocketNotifier(int socket, QSocketNotifier::Type type, QObject *parent = nullptr);
+    SafeSocketNotifier(qintptr socket, QSocketNotifier::Type type, QObject *parent = nullptr);
 
     ~SafeSocketNotifier() override;
 
@@ -41,7 +41,7 @@ public:
     {
         return sn->isEnabled();
     }
-    int socket() const
+    qintptr socket() const
     {
         return sn->socket();
     }
@@ -57,7 +57,7 @@ public Q_SLOTS:
     }
 
 Q_SIGNALS:
-    void activated(int socket);
+    void activated();
 
 private:
     QSocketNotifier *sn;
diff --git a/src/support/qpipe.cpp b/src/support/qpipe.cpp
index eb32e940..6ab9d5de 100644
--- a/src/support/qpipe.cpp
+++ b/src/support/qpipe.cpp
@@ -1075,7 +1075,7 @@ public Q_SLOTS:
 #endif
     }
 
-    void sn_read_activated(int)
+    void sn_read_activated()
     {
 #ifdef Q_OS_UNIX
         if (blockReadNotify)
@@ -1086,7 +1086,7 @@ public Q_SLOTS:
 #endif
     }
 
-    void sn_write_activated(int)
+    void sn_write_activated()
     {
 #ifdef Q_OS_UNIX
         writeResult = 0;
